### PR TITLE
Possible changes to Virtual Type List

### PR DESCRIPTION
**Pull request by subdavis**
## Summary of changes

NOTE: this change is significantly easier to read if you ignore template changes, since those are mostly just reorganization.

Changes to height calculation

* Remove expansion panels, really simplified things.
* Move height measurement into `Sidebar.vue`, v-resize, and add a prop to sub-elements telling them how tall they're allowed to be. Sub-components are responsible for figuring out the heights of their header and virtual list section.  I've hard-coded these because IMO it's simpler and no worse than tagging and measuring the DOM.
* Had to do something about the threshold bar since it's dynamic.  I added a prop to the sidebar to indicate that the threshold bar is or is not there, and another hardcoded height I'm not happy about.  But again, it's simple code, so /shrug/

Changes to type list item rows.

* I think what I did is behaviorally correct (screenshots attached), but I'm unhappy with the implementation.
* There probably just needs to be a width prop on the list row item and each part of the row needs to have a fixed with.
* I don't think you can rely on character widths (20 chars) to make this work, you need fixed width and `text-overflow: ellipsis;`

Other minor changes

Moved deletion menu settings below the creation settings, it just seemed like the right ordering.

# Before

|  Issue with row widths | Issue with resize: box should expand | Issue with resize: box has no scrollbar but overflows |
|---|---|---|
Screenshot from 2021-11-12 09-47-05 | Screenshot from 2021-11-12 11-12-37  | Screenshot from 2021-11-12 11-14-43

# After
| Type settings | Track Settings | Width fixes: Share row width between confidence indicator and type text if both exist. |
|---|---|---|
Screenshot from 2021-11-12 13-39-49 | Screenshot from 2021-11-12 13-40-00 | Screenshot from 2021-11-12 16-16-32

### CODE REVIEW

File: client/src/components/TrackList.vue [hunk 1]
```diff
@@ -221,6 +221,8 @@ export default defineComponent({
       ];
     });
 
+    const virtualHeight = computed(() => props.height - 52);
```
Comment (BryonLewis) [inline client/src/components/TrackList.vue:224]: either prop or make note that this 52 is a magic number based on the `confidenceThresholdHeight` height located in `Sidebar.vue`?

**Review by BryonLewis — APPROVED** [review]
forgot about css `text-overflow: ellipsis` and this works better without having to use `ResizeObserver`.  The only thing that's different is that type list and track list are now locked to a 45/55 split instead of having the typelist shrink/grow as necessary.

File: client/src/components/TrackList.vue (hunk 1 above)
Comment (subdavis) [reply client/src/components/TrackList.vue]: Great call.  See 73a80f -- I'm much happier with this now.  the `width` is now passed down as a prop and the max-width of the content area is a function of that width.  The CSS in `TypeItem` was simplified further to accept a runtime variable.

**Review by BryonLewis — APPROVED** [review]
Pulled and tested the latest version looks good.

**Pull request merged**